### PR TITLE
removing localhost sentry events

### PR DESCRIPTION
**Pull request by guillim**
As we discussed, we needed to figure out all events from local host, since it would pollute our dashboard. That's the objective of this PIR

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -5,24 +5,27 @@ import * as Sentry from '@sentry/angular-ivy';
...
-  // Performance Monitoring
-  tracesSampleRate: environment.sentryTracesSampleRate, //  Capture 100% of the transactions
-  // Session Replay
-  replaysSessionSampleRate: environment.sentryReplaysSessionSampleRate, // This sets the sample rate at 10%. You may want to change it to 100% while in development and then sample at a lower rate in production.
-  replaysOnErrorSampleRate: environment.sentryReplaysOnErrorSampleRate, // If you're not already sampling the entire session, change the sample rate to 100% when sampling sessions where errors occur.
-});
+if (environment.production) {
+  Sentry.init({
+    dsn: environment.sentryDsn,
+    environment: environment.sentryEnvironment === 'production' ? 'production' : 'development',
```
Comment (oliviermlm) [inline src/main.ts:11]: Since the whole block is wrapped in the if condition, some cleaning might be needed to avoid confusion

File: src/main.ts [hunk 2]
```diff
@@ -5,24 +5,27 @@ import * as Sentry from '@sentry/angular-ivy';
...
+    integrations: [
+      new Sentry.BrowserTracing({
+        // Set 'tracePropagationTargets' to control for which URLs distributed tracing should be enabled
+        tracePropagationTargets: ['localhost', /^https:\/\/getcockpit\.io/],
```
Comment (oliviermlm) [inline src/main.ts:15]: Tracing should not be enabled for localhost, in any case

File: src/environments/environment.preprod.ts [hunk 3]
```diff
@@ -11,4 +11,5 @@ export const environment = {
   sentryTracesSampleRate: 1.0,
   sentryReplaysSessionSampleRate: 0,
   sentryReplaysOnErrorSampleRate: 1.0,
+  sentryEnvironment: 'production',
```
Comment (oliviermlm) [inline src/environments/environment.preprod.ts:14]: The sentry environment is set to 'production' but it is the 'preprod' environment file

File: src/environments/environment.prod.ts [hunk 4]
```diff
@@ -11,4 +11,5 @@ export const environment = {
   sentryTracesSampleRate: 1.0,
   sentryReplaysSessionSampleRate: 0,
   sentryReplaysOnErrorSampleRate: 1.0,
+  sentryEnvironment: 'development',
```
Comment (oliviermlm) [inline src/environments/environment.prod.ts:14]: Should be set as 'production' ?

File: src/environments/environment.ts [hunk 5]
```diff
@@ -15,6 +15,7 @@ export const environment = {
   sentryTracesSampleRate: 1.0,
   sentryReplaysSessionSampleRate: 0,
   sentryReplaysOnErrorSampleRate: 1.0,
+  sentryEnvironment: 'development',
```
Comment (oliviermlm) [inline src/environments/environment.ts:18]: Not sure if necessary

**Review by oliviermlm — CHANGES_REQUESTED** [review]

File: src/main.ts (hunk 1 above)
Comment (guillim) [reply src/main.ts]: true

**Review by oliviermlm — APPROVED** [review]

**Pull request merged**